pkg/tsdb/query_endpoint.go

```diff
@@ -20,6 +20,10 @@ func init() {
 }
 
 func getTsdbQueryEndpointFor(dsInfo *models.DataSource) (TsdbQueryEndpoint, error) {
+	if dsInfo.Type == "stackdriver" { // legacy
```

Comment:
Having this logic here is a code smell. Instead of this, can we register query endpoint for both legacy stackdriver and new cloud monitoring? See my comment of plugin.json